java_tools/configuration_definition/src/com/rusefi/ConfigDefinition.java

```diff
@@ -414,75 +405,126 @@ private static void processYamlFile(File yamlFile, ReaderState state,
         SystemOut.println(data);
         Objects.requireNonNull(data, "data");
         for (Map<String, Object> pin : data) {
-            Object idObject = pin.get("id");
-            if (idObject instanceof ArrayList) {
-                ArrayList IDs = (ArrayList) idObject;
-                Object classes = pin.get("class");
-                if (!(classes instanceof ArrayList))
-                    throw new IllegalStateException("Expected multiple classes for " + IDs);
-                for (int i = 0; i < IDs.size(); i++) {
-                    String id = (String) IDs.get(i);
-                    findMatchingEnum(id,
-                            (String) pin.get("ts_name"),
-                            (String) ((ArrayList) classes).get(i),
-                            state, listOutputs, listAnalogInputs, listEventInputs, listSwitchInputs);
+            Object pinId = pin.get("id");
+            Object pinClass = pin.get("class");
+            Object pinName = pin.get("ts_name");
+            if (pinId == null || pinClass == null || pinName == null) {
+                continue;
+            }
+            if (pinId instanceof ArrayList) {
+                ArrayList pinIds = (ArrayList) pinId;
+                if (!(pinClass instanceof ArrayList))
+                    throw new IllegalStateException("Expected multiple classes for " + pinIds);
+                for (int i = 0; i < pinIds.size(); i++) {
+                    String id = (String) pinIds.get(i);
+                    Map<String, Object> thisPin = new HashMap();
+                    thisPin.put("id", pinIds.get(i));
+                    thisPin.put("ts_name", pinName);
+                    thisPin.put("class", ((ArrayList) pinClass).get(i));
+                    listPins.add(thisPin);
                 }
-            } else if (idObject instanceof String ) {
-                findMatchingEnum((String) idObject, (String) pin.get("ts_name"), (String) pin.get("class"), state, listOutputs, listAnalogInputs, listEventInputs, listSwitchInputs);
+            } else if (pinId instanceof String ) {
+                Map<String, Object> thisPin = new HashMap();
+                thisPin.put("id", pinId);
+                thisPin.put("ts_name", pinName);
+                thisPin.put("class", pinClass);
+                listPins.add(thisPin);
             }
```

Comment:
we should throw in the `else { }` case if it's neither an array nor string

Reply:
I should be able to get the class name with pinId.getClass().getCanonicalName(), right? Or do I want getSimpleName()?

Reply:
Oh or getName()

Reply:
something like that, yeah